easybuild/easyblocks/e/elpa.py

```diff
@@ -0,0 +1,227 @@
+
+    def run_all_steps(self, *args, **kwargs):
+        """
+        Put configure options in place for different builds (with and without openmp).
+        """
+
+        # make all builds verbose
+        self.cfg.update('preconfigopts', 'autoreconf &&')
+
+        # the following configopts are common to all builds
+        if self.toolchain.options['pic']:
+            self.cfg.update('configopts', '--with-pic')
+
+        if self.cfg['with_shared']:
+            self.cfg.update('configopts', '--enable-shared')
+
+        if self.cfg['with_generic_kernel']:
+            self.cfg.update('configopts', '--enable-generic')
+
+        if self.cfg['with_single']:
+            self.cfg.update('configopts', '--enable-single-precision')
+
+        for flag in ELPA_CPU_FEATURE_FLAGS:
+            # many ELPA kernels are enabled by default, even when the
+            # CPU does not support them, so we disable them all, except
+            # when the appropriate CPU flag is found
+            # sse kernels require sse4_2
+            if flag == 'sse4_2':
+                if getattr(self, flag):
+                    self.cfg.update('configopts','--enable-sse')
+                    self.cfg.update('configopts', '--enable-sse-assembly')
+                else:
+                    self.cfg.update('configopts', '--disable-sse')
+                    self.cfg.update('configopts', '--disable-sse-assembly')
+            elif flag == 'avx512f':
+                if getattr(self, 'avx512f'):
+                    self.cfg.update('configopts', '--enable-avx512')
+                else:
+                    self.cfg.update('configopts', '--disable-avx512')
+            else:
+                if getattr(self, flag):
+                    self.cfg.update('configopts', '--enable-%s' % flag)
+                else:
+                    self.cfg.update('configopts', '--disable-%s' % flag)
+
+        # By default ELPA tries to use MPI and configure fails if it's not available
+        # so we turn it off if MPI is not available
+        # We also set the LIBS environmet variable to detect the correct linalg library
+        # depending on the MPI availability.
+        if self.toolchain.options.get('usempi', None):
+            self.cfg.update('configopts', 'LIBS="$LIBSCALAPACK"')
+        else:
+            self.cfg.update('configopts', '--with-mpi=no')
+            self.cfg.update('configopts', 'LIBS="$LIBLAPACK"')
+
+        # make all builds verbose
+        self.cfg.update('buildopts', 'V=1')
+
+        # keep track of common configopts and of configopts specified
+        # in easyconfig file, so we can include them in each iteration
+        # later
+        common_config_opts = self.cfg['configopts']
+        common_build_opts = self.cfg['buildopts']
+
+        self.cfg['configopts'] = []
+        self.cfg['buildopts'] = []
+
+        with_omp_opts = [False]
+        if self.cfg['with_openmp']:
+            with_omp_opts.append(True)
```

Comment:
@boegel No, there's probably no reason. This basically came from the FFTW easyblock, as you and others suggested I use it as a starting point.

Reply:
@boegel Looking at this again, it's not clear to me what the behavior of the easyblock should be regarding OpenMP.

At the moment two versions of the library are built, with and without OpenMP support, unless  `with_openmp` is explicitly set to `False`, in which case only the version without OpenMP support is built.

Now, if I change the easyblock the way you suggest to use the `openmp` toolchain option, what should be the behavior? Should the easyblock build both versions unless `openmp` is set explicitly to `False`? This is not the usual behavior of the `openmp` toolchain option. Do you think this could be confusing for users?

Reply:
@micaeljtoliveira I don't think you can easily tell whether or not the `openmp` toolchain option was explicitly set to `False` or not, since the default value is `False` (and the `openmp` toolchain option is always defined as soon as there's a compiler component in the toolchain , so `self.toolchain.get('openmp')` will always return the default value).

I guess that answer the question: only building with OpenMP support when the `openmp` toolchain option is set to `True` makes sense.